src/status_im/chat/models/pin_message.cljs

```diff
@@ -0,0 +1,99 @@
+  [{:keys [db]} current-chat-id _ err]
+  (log/error "failed loading pin messages" current-chat-id err)
+  (when current-chat-id
+    {:db (assoc-in db [:pagination-info current-chat-id :loading-pin-messages?] false)}))
+
+(fx/defn pin-messages-loaded
+  {:events [::pin-messages-loaded]}
+  [{db :db} chat-id {:keys [cursor pinned-messages]}]
+  (let [all-messages (reduce (fn [acc {:keys [message-id] :as message}]
+                               (assoc acc message-id message))
+                             {}
+                             pinned-messages)
+        messages-id-list (map :message-id pinned-messages)]
+    {:db (-> db
+             (assoc-in [:pagination-info chat-id :loading-pin-messages?] false)
+             (assoc-in [:pin-messages chat-id] all-messages)
+             (assoc-in [:pin-message-lists chat-id] (message-list/add-many nil (vals all-messages)))
+             (assoc-in [:pagination-info chat-id :all-pin-loaded?]
+                       (empty? cursor)))}))
+
+(fx/defn receive-signal
+  [{:keys [db] :as cofx} pin-messages]
+  (let [{:keys [chat-id]} (first pin-messages)]
+    (when (= chat-id (db :current-chat-id))
+      (let [{:keys [chat-id]} (first pin-messages)
+            already-loaded-pin-messages (get-in db [:pin-messages chat-id] {})
+            already-loaded-messages (get-in db [:messages chat-id] {})
+            {:keys [all-messages new-messages]} (reduce (fn [{:keys [all-messages] :as acc}
+                                                             {:keys [message_id pinned from]
+                                                              :as   message}]
+                                                          (let [current-message (get already-loaded-messages message_id)
+                                                                current-message-pin (merge current-message
+                                                                                           {:pinned    pinned
+                                                                                            :pinned-by from})]
+                                                            (cond-> acc
+                                                              (and pinned (nil? current-message))
+                                                              (update :new-messages conj current-message-pin)
+
+                                                              (nil? pinned)
+                                                              (update :all-messages dissoc message_id)
+
+                                                              (and (some? pinned) (some? current-message))
+                                                              (update :all-messages assoc message_id current-message-pin))))
+                                                        {:all-messages already-loaded-pin-messages
+                                                         :new-messages []}
```

Comment:
1. Do we need this empty `:new-messages`? 
2. Can you please document the reduce function? It's hard to understand whats happening in a glance.

Reply:
1. We don't need it, I refactored this part and forgot to remove it. 
2. Sure, adding it right now (and also simplifying the function)